src/devices/bus/vcs/rom.cpp

```diff
@@ -624,44 +417,53 @@ void a26_rom_3f_device::write_bank(offs_t offset, uint8_t data)
 
 a26_rom_e0_device::a26_rom_e0_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
 	: a26_rom_base_device(mconfig, A26_ROM_E0, tag, owner, clock)
+	, m_bank0(*this, "bank0")
+	, m_bank1(*this, "bank1")
+	, m_bank2(*this, "bank2")
 {
 }
 
-void a26_rom_e0_device::device_start()
-{
-	save_item(NAME(m_base_banks));
-}
-
 void a26_rom_e0_device::device_reset()
 {
-	m_base_banks[0] = 4;
-	m_base_banks[1] = 5;
-	m_base_banks[2] = 6;
-	m_base_banks[3] = 7;
+	m_bank0->set_entry(4);
+	m_bank1->set_entry(5);
+	m_bank2->set_entry(6);
 }
 
 void a26_rom_e0_device::install_memory_handlers(address_space *space)
 {
-	space->install_read_handler(0x1000, 0x1fff, read8sm_delegate(*this, FUNC(a26_rom_e0_device::read)));
-	space->install_write_handler(0x1000, 0x1fff, write8sm_delegate(*this, FUNC(a26_rom_e0_device::write)));
+	m_bank0->configure_entries(0, 8, get_rom_base(), 0x400);
+	m_bank1->configure_entries(0, 8, get_rom_base(), 0x400);
+	m_bank2->configure_entries(0, 8, get_rom_base(), 0x400);
+
+	space->install_read_bank(0x1000, 0x13ff, m_bank0);
+	space->install_read_bank(0x1400, 0x17ff, m_bank1);
+	space->install_read_bank(0x1800, 0x1bff, m_bank2);
+	space->install_rom(0x1c00, 0x1fff,get_rom_base() + 7 * 0x400);
+	space->install_write_handler(0x1fe0, 0x1fe7, write8sm_delegate(*this, FUNC(a26_rom_e0_device::switch_bank0)));
+	space->install_write_handler(0x1fe8, 0x1fef, write8sm_delegate(*this, FUNC(a26_rom_e0_device::switch_bank1)));
+	space->install_write_handler(0x1ff0, 0x1ff7, write8sm_delegate(*this, FUNC(a26_rom_e0_device::switch_bank2)));
+	space->install_read_tap(0x1fe0, 0x1fe7, "bank",
+			[this] (offs_t address, u8 &, u8) { if (!machine().side_effects_disabled()) switch_bank0(address, 0); });
+	space->install_read_tap(0x1fe8, 0x1fef, "bank",
+			[this] (offs_t address, u8 &, u8) { if (!machine().side_effects_disabled()) switch_bank1(address, 0); });
+	space->install_read_tap(0x1ff0, 0x1ff7, "bank",
+			[this] (offs_t address, u8 &, u8) { if (!machine().side_effects_disabled()) switch_bank2(address, 0); });
```

Comment:
Do you actually need three trampolines for setting the banks?  You could use a `memory_bank_array_creator` for the banks and then use a single function with the bank number as a template parameter.

Reply:
Thanks for the pointer, changed to memory_bank_array_creator.